src/main/java/lt/vu/trip/controller/TripController.java

```diff
@@ -16,14 +20,59 @@
 	private TripService tripService;
 
 	@GetMapping("")
-	public ResponseEntity<List<Trip>> getAllTrips() {
-		List<Trip> trips = tripService.getAll();
-		return ResponseEntity.ok(trips);
+	public ResponseEntity<TripListResponse> getAllTrips(@Valid TripRequestParams requestParams) {
```

Comment:
TripRequestParams has all available URL query params. These params are then mapped to TripSearchCriteria, which is used in entity repository.